### PR TITLE
Convert Dapr middlewares and HTTP channel from fasthttp to net/http

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
# Description

This PR (and its [companion PR in components-contrib](https://github.com/dapr/components-contrib/pull/2143)) converts all Dapr middlewares from fasthttp to net/http (standard library).

In order for that to work, I also had to backport a change from #5170, changing the HTTP channel from using fasthttp to net/http. 

Why:

1. fasthttp as a server is going to be removed from Dapr, so the middlewares need to be converted (#4979)
2. When streaming is merged (#5170), the HTTP app channel switches from using fasthttp as client to net/http, so fasthttp middlewares can't be used (as implemented in #5132)
3. From the perspective of end users, writing middlewares for net/http is arguably simpler than writing them for fasthttp, given that it's part of the standard library and has lots of examples and existing libraries

## Issue reference

Part of #4979 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Unit tests passing
* [x] End-to-end tests passing

### CODE REVIEW

File: utils/nethttpadaptor/nethttpadaptor.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/*
```
Comment (ItalyPaleAle) [inline utils/nethttpadaptor/nethttpadaptor.go:1]: This file was previously in components-contrib, where it’s not needed anymore. Moved here as it’s needed until fasthttp is completely removed

**Comment by daixiang0** [comment]
Where to see the details of perf tests?

**Comment by yaron2** [comment]
> Where to see the details of perf tests? I am concerned about the performance changes.

https://github.com/dapr/dapr/actions/runs/3208910024/jobs/5245220552

**Comment by ItalyPaleAle** [comment]
Note that our perf tests don't cover middlewares.

(Also, this would add better reporting for the bot: https://github.com/dapr/dapr/pull/5307 )

**Comment by ItalyPaleAle** [comment]
Although our perf tests do cover the HTTP channel so I stand correct and they still matter a lot :)

**Review by artursouza — APPROVED** [review]

**Comment by yaron2** [comment]
Reverting this PR as it caused a major regression in perf. (Service invocation p90 added latency went up from ~1.07ms to > +1.70ms).

File: pkg/channel/http/http_channel.go [hunk 2]
```diff
@@ -75,12 +77,12 @@ func CreateLocalChannel(port, maxConcurrency int, pipeline httpMiddleware.Pipeli
...
 	c := &Channel{
 		pipeline: pipeline,
-		client: &fasthttp.Client{
-			MaxConnsPerHost:           1000000,
-			MaxIdemponentCallAttempts: 0,
-			MaxResponseBodySize:       maxRequestBodySize * 1024 * 1024,
-			ReadBufferSize:            readBufferSize * 1024,
-			DisablePathNormalizing:    true,
+		// We cannot use fasthttp here because of lack of streaming support
+		client: &http.Client{
```
Comment (yaron2) [inline pkg/channel/http/http_channel.go:81]: @ItalyPaleAle I believe that setting `MaxConnsPerHost` on the http.Client will improve performance by a wide margin (you need to create a `DefaultTransport`).

**Comment by ItalyPaleAle** [comment]
@yaron2  can you please re-open this so I can continue working on it?

File: pkg/channel/http/http_channel.go (hunk 2 above)
Comment (ItalyPaleAle) [reply pkg/channel/http/http_channel.go:81]: Thanks, this sounds like solid advice

**Comment by yaron2** [comment]
> @yaron2 can you please re-open this so I can continue working on it?

I can't, as you deleted the branch.

**Comment by ItalyPaleAle** [comment]
Take 2 on #5330. Will continue working there.

**Pull request merged**